packages/compiler/src/emitter-framework/asset-emitter.ts

```diff
@@ -425,6 +423,15 @@ export function createAssetEmitter<T, TOptions extends object>(
     emitTupleLiteralValues(tuple) {
       return invokeTypeEmitter("tupleLiteralValues", tuple);
     },
+
+    emitSourceFile(sourceFile) {
+      if (emittedSourceFiles.has(sourceFile)) {
```

Comment:
This isn't strictly needed but seemed good to have - if you happen to call emitSourceFile on a source file that has already been emitted, just return the already-emitted thing.

Reply:
is that maybe not the sign that there is a bug if you try to write the same file twice?

Reply:
Not necessarily. emitSourceFile's job isn't to write the file but to convert the built up emit structure into actual bytes to write. It is conceivable that you might call this purposely - I added this because at one point I was doing that - calling emitFile to get the json schema contents of the bundled refs - before I found a better way. That said, I'm not attached to this and am happy to revert to the old behavior for now.

Reply:
I see the potential use case.

could there be a case where you call `emitSourceFile` then change something in the emit structure and call again? or this is not likely to be happening.